src/Shared/ParameterBindingMethodCache.cs

```diff
@@ -281,7 +285,8 @@ static bool ValidateReturnType(MethodInfo methodInfo)
             {
                 if (candidateInterfaceMethodInfo is not null)
                 {
-                    throw new InvalidOperationException($"{TypeNameHelper.GetTypeDisplayName(type, fullName: false)} implements multiple interfaces defining a static {interfaceMethod} method causing ambiguity.");
+                    return !_throwOnInvalidMethod ? null :
+                        throw new InvalidOperationException($"{TypeNameHelper.GetTypeDisplayName(type, fullName: false)} implements multiple interfaces defining a static {interfaceMethod} method causing ambiguity.");
```

Comment:
Consider rewriting this as an if for clarity
```suggestion
                    if (_throwOnInvalidMethod)
                    {
                        throw new InvalidOperationException($"{TypeNameHelper.GetTypeDisplayName(type, fullName: false)} implements multiple interfaces defining a static {interfaceMethod} method causing ambiguity.");
                    }
                    
                    return null;
```